src/raftstore/store/peer.rs

```diff
@@ -505,7 +510,20 @@ impl Peer {
             try!(self.send(trans, ready.messages.drain(..), &mut metrics.message));
         }
 
-        let apply_result = try!(self.mut_store().handle_raft_ready(&ready));
+        let apply_result = match self.mut_store().handle_raft_ready(&ready) {
+            Ok(r) => r,
+            Err(e) => {
+                // Because we send the messages before handle raft ready when self
+                // is leader so if we don't panic here, the entries in memory may
+                // be committed before next handle raft ready. If next handle raft
+                // ready still failed to write rocksdb, then entries are committed
```

Comment:
Could get the point of all comments here, and the verb tenses seem messy.
A simple description should appear as the first statement of a bunk of comments. Like
"It must panic when leader fail to persist committed entries. And ...(why is that)"